### PR TITLE
ENG-1524: Change retry mechansim default in LightningClient

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes [ENG-1524](https://lightningai.atlassian.net/browse/ENG-1524)

Previously, calls to the LightningClient would follow an exponential backoff retry mechanism when calling APIs. This is desired for calls in the cloud, but not for calls in the CLI. This PR changes this default. The cloud backend will still use the retry mechanism, but the LightningClient instances used in the CLI will immediately error out when an API call can't reach the endpoint.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda

### CODE REVIEW

**Comment by awaelchli** [comment]
The test `tests_app/utilities/test_secrets.py::test_names_to_ids` is failing here in the CI but not locally. I suspect we set an environment variable somewhere that I don't have set locally, but I don't know where. I can't relate it to the changes in this PR. @tchaton @manskx any idea what it could be?

**Review by nmiculinic — APPROVED** [review]
Thank you!

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by awaelchli** [comment]
Blocked by issues in meta package replacing imports #15446

File: src/lightning_app/utilities/cloud.py [hunk 1]
```diff
@@ -11,14 +11,16 @@
 
 def _get_project(client: LightningClient, project_id: str = LIGHTNING_CLOUD_PROJECT_ID) -> V1Membership:
     """Get a project membership for the user from the backend."""
+    print(client.projects_service_list_memberships)
+    assert False
```
Comment (manskx) [inline src/lightning_app/utilities/cloud.py:15]: could be removed ?

Comment (awaelchli) [reply src/lightning_app/utilities/cloud.py:15]: I'm still debugging. The ****** CI won't let this test pass. The mock patch does not work in the CI but it works locally.

File: src/lightning_app/utilities/cloud.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def _get_project(client: LightningClient, project_id: str = LIGHTNING_CLOUD_PROJ
                 break
         else:
             raise ValueError(
-                "Environment variable LIGHTNING_CLOUD_PROJECT_ID is set " "but could not find an associated project."
+                "Environment variable LIGHTNING_CLOUD_PROJECT_ID is set but could not find an associated project."
```
Comment (Borda) [inline src/lightning_app/utilities/cloud.py:21]:
```suggestion
                "Environment variable `LIGHTNING_CLOUD_PROJECT_ID` is set but could not find an associated project."
```

**Review by Borda — APPROVED** [review]

**Comment by justusschock** [comment]
@awaelchli mind have a look at failing tests? Could be due to secrets?

**Pull request merged**